docs/docs-beta/docs/guides/deployment/kubernetes.md

```diff
@@ -1,4 +1,220 @@
 ---
 title: "Deploy to Kubernetes"
 sidebar_position: 21
----
\ No newline at end of file
+---
+
+Deploying Dagster to a Kubernetes cluster is a popular option for maintaining a production Dagster deployment. Dagster uses Helm, a package manager for Kubernetes applications, to help manage deploying Dagster to a Kubernetes cluster.
+
+Dagster provides [Helm charts](https://github.com/dagster-io/dagster/tree/master/helm) for deploying Dagster that you can customize for your specific needs. For each Dagster component used by the Helm chart, Dagster publishes a corresponding image to [DockerHub](https://hub.docker.com/u/dagster).
+
+## What you'll learn
+
+- How to get your Dagster project ready to deploy to your Kubernetes cluster
+- How to build a Docker image containing your Dagster project to deploy to your cluster
+- How to access and customize the Dagster-provided Helm charts
+- How to apply the configuration in the Helm charts to your Kubernetes cluster to deploy your Dagster project
+- How to connect to your deployed Dagster project
+
+<details>
+  <summary>Prerequisites</summary>
+
+To follow the steps in this guide, you'll need:
+
+- **Familiarity with [Docker](https://docs.docker.com/)**, and:
+  - **To have Docker installed**. [Docker installation guide](https://docs.docker.com/engine/install/)
+  - **Access to a Docker image registry**, such as Amazon Web Services ECR or DockerHub. If you're following along on your local machine, this isn't required.
+- **Familiarity with [Kubernetes](https://kubernetes.io/docs/home/)**, and:
+  - **To have `kubectl` installed**. [Kubernetes installation guide](https://kubernetes.io/docs/tasks/tools/)
+  - **An existing Kubernetes cluster**. To follow along on your local machine, [install Docker Desktop](https://docs.docker.com/desktop/kubernetes/) and turn on the included Kubernetes server.
+- **Familiarity with [Helm](https://helm.sh/docs/)**, and:
+  - **To have Helm 3 installed**. [Helm installation guide](https://helm.sh/docs/intro/install/)
+- A Dagster project to deploy. You can also use the [example project](/todo):
+  ```bash
+  dagster project from-example --example deploy_k8s_beta --name deploy_k8s_beta
+  ```
+
+</details>
+
+## Step 1: Prepare your Dagster project
+You will need to add a `workspace.yaml` file to your Dagster project to be ready to deploy with Kubernetes. If you are using the [example project](/todo), you will already have this file.
```

Comment:
you might need a workspace.yaml to run locally, but you don't need it to deploy to kubernetes. the helm chart creates a workspace.yaml that the k8s cluster uses based on the information under dagster-user-deployments

Reply:
oh sick, i didn't know that!